crates/bevy_tasks/src/task_pool.rs

```diff
@@ -265,7 +266,7 @@ impl Default for TaskPool {
 pub struct Scope<'scope, T> {
     executor: &'scope async_executor::Executor<'scope>,
     local_executor: &'scope async_executor::LocalExecutor<'scope>,
-    spawned: Vec<async_executor::Task<T>>,
+    spawned: Arc<Mutex<Vec<async_executor::Task<T>>>>,
```

Comment:
Replacing this Arc with a reference to a `Mutex<Vec<T>>` stored on the stack should be possible

Reply:
Stack of which thread?

Reply:
It'd be similar to how the code currently makes a fresh `Executor` on the stack when `scope` is called. We'd just also make a `Mutex<Vec<Task<T>>>` on the stack, and store a regular rust reference to it inside `Scope`